bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaCurtainHandler.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.somfytahoma.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+
+import static org.openhab.binding.somfytahoma.internal.SomfyTahomaBindingConstants.*;
+
+/**
+ * The {@link SomfyTahomaCurtainHandler} is responsible for handling commands,
+ * which are sent to one of the channels of the roller shutter, screen & garage door
+ * things.
+ *
+ * @author Tobias Ammann - Initial contribution
+ */
+@NonNullByDefault
+public class SomfyTahomaCurtainHandler extends SomfyTahomaBaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyTahomaCurtainHandler.class);
+
+    public SomfyTahomaCurtainHandler(Thing thing) {
+        super(thing);
+        stateNames = new HashMap<String, String>() {{
+            put(CONTROL, "core:ClosureState");
+        }};
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received command {} for channel {}", command, channelUID);
+        if (!CONTROL.equals(channelUID.getId())) {
```

Comment:
What about to 
1) make the method ```getTahomaCommand``` in the ```SomfyTahomaRollerShutterHandler``` class protected 
2) inherit ```SomfyTahomaCurtainHandler``` from ```SomfyTahomaRollerShutterHandler``` and remove ```handleCommand``` method
3) add ```@override``` to ```getTahomaCommand``` in ```SomfyTahomaCurtainHandler```